server/src/main/java/io/druid/server/StatusResource.java

```diff
@@ -21,25 +21,47 @@
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.Maps;
 import com.sun.jersey.spi.container.ResourceFilters;
 import io.druid.initialization.DruidModule;
 import io.druid.initialization.Initialization;
 import io.druid.java.util.common.StringUtils;
 import io.druid.server.http.security.StateResourceFilter;
 
+import javax.inject.Inject;
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
+import java.util.Map;
+import java.util.Properties;
 
 /**
  */
 @Path("/status")
 public class StatusResource
 {
+
+  private final Properties properties;
+
+  @Inject
+  public StatusResource(Properties properties)
+  {
+    this.properties = properties;
+  }
+
+  @GET
+  @Path("/properties")
+  @ResourceFilters(StateResourceFilter.class)
```

Comment:
this seems to be a config state filter? @pjain1 is it okay ?

Reply:
Yes, looks more like a read on config. Please use `ConfigResourceFilter` and add `requestPath.startsWith("status/properties")` to the `isApplicable` method of the class.